aea/test_tools/test_cases.py

```diff
@@ -0,0 +1,289 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This module contains test case classes based on pytest for AEA end-to-end testing."""
+
+import os
+import shutil
+import subprocess  # nosec
+import sys
+import tempfile
+from pathlib import Path
+
+import pytest
+
+from aea.cli import cli
+from aea.configurations.base import PublicId
+from aea.mail.base import Envelope
+from aea.protocols.default.message import DefaultMessage
+from aea.protocols.default.serialization import DefaultSerializer
+from aea.test_tools.exceptions import AEATestingException
+
+from tests.common.click_testing import CliRunner
+from tests.conftest import AUTHOR, CLI_LOG_OPTION
+
+
+class AEATestCase:
+    """Test case for AEA end-to-end tests."""
+
+    @classmethod
+    def setup_class(cls):
+        """Set up the test class."""
+        cls.runner = CliRunner()
+        cls.cwd = os.getcwd()
+        cls.t = tempfile.mkdtemp()
+
+        # add packages folder
+        packages_src = os.path.join(cls.cwd, "packages")
```

Comment:
"packages" should be an argument, not hardcoded. Can be a default argument, e.g. `setup_class(cls, packages_dir_path = DEFAULT_PACKAGES_DIR_PATH)`